tools/build-release-branch.sh

```diff
@@ -16,117 +17,133 @@
-        echo ""
-        # reset --hard to remote master in case they have local commits in their repo
-        git checkout master && git pull && git reset --hard origin/master
-
-        # Create new branch, push to repo
-        git checkout -b $NEW_UNBUILT_BRANCH
-
-        git push -u origin $NEW_UNBUILT_BRANCH
-        echo ""
-        echo "$NEW_UNBUILT_BRANCH created."
-        echo ""
-        # Verify you want a built version
-        read -n1 -p "Would you like to create a built version of $NEW_UNBUILT_BRANCH as new $NEW_BUILT_BRANCH? [y/N]" reply
-        if [[ 'y' == $reply || 'Y' == $reply ]]
-        then
-            # make sure we're still checked out on the right branch
-            git checkout $NEW_UNBUILT_BRANCH
-
-            git checkout -b $NEW_BUILT_BRANCH
-
-            # New .gitignore for release branches
-            echo ""
-            echo "Creating new .gitignore"
-            echo ""
-            create_release_gitignore
-
-            # Remove stuff from svnignore
-            modify_svnignore
-
-            git checkout $NEW_UNBUILT_BRANCH
-
-            git push -u origin $NEW_BUILT_BRANCH
-
-            # Script will continue on to actually build the plugin onto this new branch...
-        else
-            # Nothing left to do...
-            echo ""
-            echo "Ok, all done then."
-            exit 1
-        fi
-    fi
+	# Prompt for version number.
+	read -p "What version are you releasing? Please write in x.x syntax. Example: 4.9 - " version
+
+	# Declare the new branch names.
+	NEW_UNBUILT_BRANCH="branch-$version"
+	NEW_BUILT_BRANCH="branch-$version-built"
+
+	# Check if branch already exists, if not, create new branch named "branch-x.x"
+	if [[ -n $( git branch -r | grep "$NEW_UNBUILT_BRANCH" ) ]]; then
+		echo "$NEW_UNBUILT_BRANCH already exists.  Exiting..."
+		exit 1
+	else
+		echo ""
+		echo "Creating new unbuilt branch $NEW_UNBUILT_BRANCH from current master branch..."
+		echo ""
+		# reset --hard to remote master in case they have local commits in their repo
+		git checkout master && git pull && git reset --hard origin/master
+
+		# Create new branch, push to repo
+		git checkout -b $NEW_UNBUILT_BRANCH
+
+		git push -u origin $NEW_UNBUILT_BRANCH
+		echo ""
+		echo "$NEW_UNBUILT_BRANCH created."
+		echo ""
+		# Verify you want a built version
+		read -n1 -p "Would you like to create a built version of $NEW_UNBUILT_BRANCH as new $NEW_BUILT_BRANCH? [y/N]" reply
+		if [[ 'y' == $reply || 'Y' == $reply ]]; then
+			# make sure we're still checked out on the right branch
+			git checkout $NEW_UNBUILT_BRANCH
+
+			git checkout -b $NEW_BUILT_BRANCH
+
+			# New .gitignore for release branches
+			echo ""
+			echo "Creating new .gitignore"
+			echo ""
+			create_release_gitignore
+
+			# Remove stuff from svnignore
+			modify_svnignore
+
+			git rm -r --cached .
+			git add .
+			git commit -m ".gitignore cleanup"
```

Comment:
Maybe make the commit msg more explicit here? Like `Remove files based on .gitignore contents`?